rts/at/sra/src/main/java/io/narayana/sra/demo/api/FlightController.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package io.narayana.sra.demo.api;
+
+import org.jboss.jbossts.star.annotation.SRA;
+import org.jboss.jbossts.star.client.SRAParticipant;
+import io.narayana.sra.demo.model.Booking;
+import io.narayana.sra.demo.service.FlightService;
+import org.jboss.jbossts.star.client.SRAStatus;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.DefaultValue;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.HeaderParam;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+
+import static org.jboss.jbossts.star.client.SRAClient.RTS_HTTP_CONTEXT_HEADER;
+
+@RequestScoped
+@Path(FlightController.FLIGHT_PATH)
+@SRA(SRA.Type.SUPPORTS)
+public class FlightController extends SRAParticipant {
+    public static final String FLIGHT_PATH = "/flight";
+    public static final String FLIGHT_NUMBER_PARAM = "flightNumber";
+    public static final String ALT_FLIGHT_NUMBER_PARAM = "altFlightNumber";
+    public static final String FLIGHT_SEATS_PARAM = "flightSeats";
+
+    @Inject
+    FlightService flightService;
+
+    @POST
+    @Path("/book")
+    @Produces(MediaType.APPLICATION_JSON)
+    @SRA(SRA.Type.REQUIRED)
+    public Booking bookFlight(@HeaderParam(RTS_HTTP_CONTEXT_HEADER) String sraId,
+                              @QueryParam(FLIGHT_NUMBER_PARAM) @DefaultValue("") String flightNumber,
+                              @QueryParam(FLIGHT_SEATS_PARAM) @DefaultValue("1") Integer seats,
+                              @QueryParam("mstimeout") @DefaultValue("500") Long timeout) {
+
+        return flightService.book(sraId, flightNumber, seats);
+    }
+
+    @GET
+    @Path("/info/{bookingId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @SRA(SRA.Type.SUPPORTS)
+    public Booking getBooking(@PathParam("bookingId") String bookingId) {
+        return flightService.get(bookingId);
+    }
+
+    @Override
+    protected SRAStatus updateParticipantState(SRAStatus status, String bookingId) {
+        System.out.println("SRA: %s: Updating flight participant state to: %s %n", bookingId, status);
```

Comment:
I doubt at the moment that is working. But since this PR is on Hold I think there is still some work to do. I personally would write something like : `System.out.println("SRA: " + bookingId + ": Updating flight participant state to: " + status);`

Reply:
`printf` performs better and is more readable for modern programmers, but I do agree that mixing `println` and `printf `style formatting won't work.

The PR was put on Hold pending resolution of JBTM-3489 which is now resolved so when Moulalis is ready for this PR to be reviewed he can remove the Hold label.

Reply:
It should be printf I changed it and %n is for new line. Hold label is removed and it available for review.
Thanks.

Reply:
space removed.